third_party/utils/src/test/io/bazel/rulesscala/utils/Scala3CompilerUtils.scala

```diff
@@ -0,0 +1,52 @@
+package io.bazel.rulesscala.utils
+
+import dotty.tools.dotc.Compiler
+import dotty.tools.dotc.core.Contexts._
+import dotty.tools.dotc.reporting.Diagnostic
+import dotty.tools.dotc.util.SourceFile
+
+import java.nio.file.Paths
+
+object Scala3CompilerUtils {
+  def runCompiler(
+    code: String,
+    extraClasspath: List[String] = List.empty,
+  ): List[Diagnostic] = {
+    val reporter = new TestReporter()
+
+    implicit val context: FreshContext = (new ContextBase).initialCtx.fresh.setReporter(reporter)
+
+    val fullClassPath =
+      builtinClasspaths.filterNot(_.isEmpty) ++
+        extraClasspath :+
+        context.settings.classpath.value
+
+    context.setSetting(context.settings.classpath, fullClassPath.mkString(":"))
+
+    val compiler = new Compiler()
+    val run = compiler.newRun
+    run.compileSources(List(SourceFile.virtual("code.scala", code, maybeIncomplete = false)))
```

Comment:
nit: `SourceFile.virtual("code.scala", code, maybeIncomplete = false) :: Nil` is more efficient than `List(...)` because `List` has to allocate an intermediate Seq of the args.

more important: `code.scala` looks a bit bland and I wonder if we will forget where it came from. I wonder if it should be passed in with a default of `scala_compiler_util_run_code.scala` or something.

Reply:
Updated as per your suggestion.